### PR TITLE
Correctly determine if a local group exists.

### LINKED ISSUE #58619: Group module not respecting "local: yes"
##### SUMMARY

Trying to to create a local group entry with "local:yes" even if entry exists in the global LDAP directory. Looks like issue is with the group module as its not respecting "local: yes"

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

group
##### ANSIBLE VERSION

```paste below
# ansible --version
ansible 2.8.1
  config file = /root/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible
  python version = 2.7.5 (default, Jun 11 2019, 12:19:05) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]

```
##### OS / ENVIRONMENT

Red Hat Enterprise Linux 7

##### STEPS TO REPRODUCE

1] system should be member of the LDAP domain, where the group nginx already exist.
2] Then execute the below playbook.

```yaml
  - name: Create /etc/group entry for nginx
    group:
      gid: 38664
      local: yes
      name: nginx
      state: present
  
  - name: Create /etc/passwd entry for nginx
    user:
      comment: nginx
      group: nginx
      home: /var/lib/nginx
      local: yes
      name: nginx
      password_lock: yes
      shell: /sbin/nologin
      state: present
      uid: 38664
```

##### EXPECTED RESULTS

it should ignore the LDAP directory and create the /etc/group local entry with mentioned GID.

##### ACTUAL RESULTS

Its creating group with different GID. In this example playbook in passing GID 38664 but output shows its created with GID 1000 as below 

```
TASK [Create /etc/group entry for nginx]
[server.example.com] => {
    "changed": false, 
    "gid": 38664, 
    "invocation": {
        "module_args": {
            "gid": 38664, 
            "local": true, 
            "name": "nginx", 
            "non_unique": false, 
            "state": "present", 
            "system": false
        }
    }, 
    "name": "nginx", 
    "state": "present", 
    "system": false
}

TASK [Create /etc/passwd entry for nginx]

changed: [server.example.com] => {
    "changed": true, 
    "comment": "nginx", 
    "create_home": true, 
    "group": 1000,      <------------------------------------------------------
    "home": "/var/lib/nginx", 
    "invocation": {
        "module_args": {
            "append": false, 
            "authorization": null, 
            "comment": "nginx", 
            "create_home": true, 
            "expires": null, 
            "force": false, 
            "generate_ssh_key": null, 
            "group": "nginx", 
            "groups": null, 
            "hidden": null, 
            "home": "/var/lib/nginx", 
            "local": true, 
            "login_class": null, 
            "move_home": false, 
            "name": "nginx", 
            "non_unique": false, 
            "password": null, 
            "password_lock": true, 
            "profile": null, 
            "remove": false, 
            "role": null, 
            "seuser": null, 
            "shell": "/sbin/nologin", 
            "skeleton": null, 
            "ssh_key_bits": 0, 
            "ssh_key_comment": "ansible-generated on server.example..com", 
            "ssh_key_file": null, 
            "ssh_key_passphrase": null, 
            "ssh_key_type": "rsa", 
            "state": "present", 
            "system": false, 
            "uid": 38664, 
            "update_password": "always"
        }
    }, 
    "name": "nginx", 
    "shell": "/sbin/nologin", 
    "state": "present", 
    "system": false, 
    "uid": 38664
}

```

```paste below

```

### PR DESCRIPTION
**Pull request by rpluem-vf**
##### SUMMARY
Fixes #58619 with the approach proposed by samdoran.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
group

### CODE REVIEW

**Comment by rpluem-vf** [comment]
ready_for_review

**Comment by rpluem-vf** [comment]
ready_for_review

**Comment by rpluem-vf** [comment]
Ready_for_review

**Comment by rpluem-vf** [comment]
Ready_for_review

**Comment by rpluem-vf** [comment]
Anything I can do to move this forward?

**Comment by rpluem-vf** [comment]
Ready_for_review

**Comment by rpluem-vf** [comment]
Anything I can do to move this PR forward?

**Comment by rpluem-vf** [comment]
Ready_for_review

**Review by samdoran — CHANGES_REQUESTED** [review]
Sorry for not reviewing this sooner. I am not really happy with how this was solved in the `user` module and this seems to be further spreading that problematic solution. That being said, I can't think of a better way to solve it. 😞 

Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment) and add integration tests and we can merge this. See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

**Comment by samdoran** [comment]
Note to future @samdoran: I told you so. 😉

**Comment by rpluem-vf** [comment]
I added a changelog fragment in d4176f279f2b25c326adea9c5701826275746e2e, but I have trouble to create a test. For a test I would need to create a group in a non local repository e.g. LDAP first and then try to create this group locally with the same gid. I am not sure how I should do this in an agnostic way such that the testing can be run at least on any Linux or even CentOS system.

**Comment by rpluem-vf** [comment]
Ready_for_review

File: changelogs/fragments/59772-fix_ansible_issue_58619.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+bugfixes:
+  - >
+     group - The group module was not correctly detecting whether a local
+     group is existing or not with local set to yes if the same group
+     exists in a non local group repository e.g. LDAP. Fix issue #58619.
```
Comment (samdoran) [inline changelogs/fragments/59772-fix_ansible_issue_58619.yaml]:
```suggestion
     exists in a non local group repository e.g. LDAP. (https://github.com/ansible/ansible/issues/58619)
```

**Comment by samdoran** [comment]
For the integration test, it's more about exercising the code path that handles the local file parsing. See the [user integration tests](https://github.com/ansible/ansible/blob/519672a6966aa5c187569d72d0bb5a3a510135eb/test/integration/targets/user/tasks/main.yml#L918-L918) for an example.

**Review by samdoran — APPROVED** [review]

**Pull request merged**